tests/docker/ducker-ak

```diff
@@ -185,6 +185,12 @@ must_popd() {
     popd &> /dev/null || die "failed to popd"
 }
 
+echo_and_do() {
+    local cmd="${@}"
+    echo "${cmd}"
+    ${cmd}
```

Comment:
The logic through `must_do`, which we were doing previously also redirects output and checks the command return. Should we do something similar?

Reply:
Well... I made this new function because the other one was checking the output and exiting if it was nonzero. That didn't allow me to print the error message I wanted. So doing the same thing here would defeat the point :)

If bash shell was a better language, I could add a lambda argument to the existing function and have the lambda return what to print on failure. But shell doesn't really do closures, so it would turn into kind of a mess.

Reply:
Fair enough.

Reply:
Also I guess some people would say that I'm doing things too much the Java way here, and that the kewl bash way of doing this would be 
```
set -x
<run command>
set +x
```
But it still feels kind of like a mess since, does `set +x` change `$?` ?
Sigh.